### ISSUE TITLE
Recaptcha slow & often incorrect

### ISSUE THREAD

**Issue by Laululytia** [maximedrn/opensea-automatic-bulk-upload-and-sale#197]
Thank you Maximedrn, after your last upgrade script works perfect upload + sell! Only one issue... Recaptcha solves pictures very long and often incorrectly. It takes like 5-6 minutes until one item is created.  Sometimes it fails and script starts to create next one.

Do you have any suggestions? 

Windows 10 pro
Radeon rx480
Intel i5-10400f
Chrome

**Comment by Kisnobeats** [comment]
> Thank you Maximedrn, after your last upgrade script works perfect upload + sell! Only one issue... Recaptcha solves pictures very long and often incorrectly. It takes like 5-6 minutes until one item is created. Sometimes it fails and script starts to create next one.
> 
> Do you have any suggestions?
> 
> Windows 10 pro Radeon rx480 Intel i5-10400f Chrome

I believe that is how it runs. I don't think it will get every single reCAPTCHA every time. This is why there is a data log for those that are missed during the process. That way, you can run it again with that file, and it will go through the process again, but with the missed ones. Rinse repeat. I am sure there will be updates in the future, but this is a huge step forward! Thank you Maximedrn.

**Comment by Laululytia** [comment]
> > Thank you Maximedrn, after your last upgrade script works perfect upload + sell! Only one issue... Recaptcha solves pictures very long and often incorrectly. It takes like 5-6 minutes until one item is created. Sometimes it fails and script starts to create next one.
> > Do you have any suggestions?
> > Windows 10 pro Radeon rx480 Intel i5-10400f Chrome
> 
> I believe that is how it runs. I don't think it will get every single reCAPTCHA every time. This is why there is a data log for those that are missed during the process. That way, you can run it again with that file, and it will go through the process again, but with the missed ones. Rinse repeat. I am sure there will be updates in the future, but this is a huge step forward! Thank you Maximedrn.

I don't mind run again those are missed, issue is bot solving each recaptcha very long, to upload and sell 1 NFT takes at least 5 min. So 2000 NFT gonna be uploading for like 166 hours. Maybe I am doing something wrong it takes so much time for every single one... ??

**Comment by Kisnobeats** [comment]
> > > Thank you Maximedrn, after your last upgrade script works perfect upload + sell! Only one issue... Recaptcha solves pictures very long and often incorrectly. It takes like 5-6 minutes until one item is created. Sometimes it fails and script starts to create next one.
> > > Do you have any suggestions?
> > > Windows 10 pro Radeon rx480 Intel i5-10400f Chrome
> > 
> > I believe that is how it runs. I don't think it will get every single reCAPTCHA every time. This is why there is a data log for those that are missed during the process. That way, you can run it again with that file, and it will go through the process again, but with the missed ones. Rinse repeat. I am sure there will be updates in the future, but this is a huge step forward! Thank you Maximedrn.
> 
> I don't mind run again those are missed, issue is bot solving each recaptcha very long, to upload and sell 1 NFT takes at least 5 min. So 2000 NFT gonna be uploading for like 166 hours. Maybe I am doing something wrong it takes so much time for every single one... ??

Nope, nothing wrong is done. It takes just as long for me as well. I am certain it's how the bot runs. It will take me about a week and a half to complete a collection of 3k NFTs. Doing this will most likely have me start making smaller 1k collections since it's very slow.

**Comment by maximedrn** [comment]
The reCAPTCHA solver is not perfect. It uses pre-trained files and therefore can make some parsing errors.
By the way, my code is not perfect, I need to add pre-trained files for crosswalks, bridges and chimneys.
Otherwise, I can't do anything for the time process, my computer solves the reCAPTCHA is like 30 seconds (5 seconds per challenge -> ~6 challenges per NFT). I think you are using your CPU (RX480 does not support CUDA), that's why it takes time.

Translated with www.DeepL.com/Translator (free version)

**Comment by poc1127** [comment]
> The reCAPTCHA solver is not perfect. It uses pre-trained files and therefore can make some parsing errors. By the way, my code is not perfect, I need to add pre-trained files for crosswalks, bridges and chimneys. Otherwise, I can't do anything for the time process, my computer solves the reCAPTCHA is like 30 seconds (5 seconds per challenge -> ~6 challenges per NFT). I think you are using your CPU (RX480 does not support CUDA), that's why it takes time.

I get this error when I run the program

question1
what is this error? it says a file is missing.

![image](https://user-images.githubusercontent.com/100990701/159687408-0c45948e-a4cb-4a22-81a4-bd0226621c52.png)

question2
how slow is recaptcha solver for you guys? it just doesn't seem to do anything. When I click some images, it did recognize 2 images of cars. But after than I had to click myself
When I run the program, the recaptcha image e.g. crosswalk pops up but nothing happens. 

question3
Only when I solve recaptcha myself, it goes over to sell function but fails because my date format is not correct or something BUT my date format is correct. It's in this format:      "duration": ["23-03-2022 21:00", "04-09-2022 21:00"],    

Not sure what the problem is?

![image](https://user-images.githubusercontent.com/100990701/159686530-c6e82b6e-2d0e-46fa-9614-3ebac25db61f.png)

**Comment by maximedrn** [comment]
> question1
> what is this error? it says a file is missing. I get this error after I enter python main.py

This is a requirements.txt file, don't worry.

> question3
> Only when I solve recaptcha myself, it goes over to sell function but fails because my date format is not correct or something BUT my date format is correct. It's in this format: "duration": ["23-03-2022 21:00", "04-09-2022 21:00"],

Your JSON file is badly organized, it should be in the same order as the `data/` files.

**Comment by Laululytia** [comment]
> Radeon

Is it possible to change some lines in your script and use alternative cude for radeon? for example TensorFlow, Caffe, MXnet ant etc. Thank you very much! You already helped a lot 🥇
